src/contracts/mixins/SplitContractMixin.sol

```diff
@@ -0,0 +1,67 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity ^0.8.27;
+
+abstract contract SplitContractMixin {
+    /// @notice The address of the second half of the contract.
+    address internal immutable viewImplementation;
+
+    constructor(
+        address _viewImplementation
+    ) {
+        viewImplementation = _viewImplementation;
+    }
+
+    /**
+     * @dev Delegates the current call to `implementation`.
+     *
+     * This function does not return to its internal call site, it will return directly to the external caller.
+     *
+     * Copied from OpenZeppelin Contracts v4.9.0 (proxy/Proxy.sol).
+     */
+    function _delegate(
```

Comment:
Instead of us defining this here, should we use this from the OZ library directly (Mainly to prevent any mistakes on our end)? Thoughts?

I can see it's being used in `_delegateView` below